sigoptlite/sigoptlitetest/endpoints/test_best_assignments.py

```diff
@@ -127,108 +143,150 @@ def test_multimetric_thresholds(self, threshold_y1, threshold_y2):
+    best_assignments_list = list(best_assignments.iterate_pages())
+    assert len(best_assignments_list) == 1
+    best_observation = best_assignments_list[0]
+    assert best_observation.values[0].value >= threshold
 
-    best_observation = best_observations_from_logger[0]
-    assert best_observation["values"][0]["value"] >= threshold
+    observations = self.conn.experiments(e.id).observations().fetch()
+    observations_list = list(observations.iterate_pages())
+    optimized_values_within_threshold = []
+    for observation in observations_list:
+      if observation.values[0].value >= threshold:
+        optimized_values_within_threshold.append(observation.values[1].value)
 
-    valid_observations = best_assignments_logger.filter_valid_full_cost_observations(observations)
-    expected_valid_observations = [o.get_client_observation(experiment) for o in valid_observations]
-    for observation in expected_valid_observations:
-      assert best_observation["values"][1]["value"] <= observation["values"][1]["value"]
+    for optimized_value in optimized_values_within_threshold:
+      assert best_observation.values[1].value <= optimized_value
 
   def test_multimetric_search(self):
     experiment_meta = self.get_experiment_feature("search")
+    experiment_meta["type"] = "random"
     thresholds = [m["threshold"] for m in experiment_meta["metrics"]]
     objectives = [m["objective"] for m in experiment_meta["metrics"]]
-    experiment = LocalExperimentBuilder(experiment_meta)
-    observations = self.make_random_observations(experiment, num_observations=25)
+    e = self.conn.experiments().create(**experiment_meta)
+
+    num_observations = 25
+    for _ in range(num_observations):
+      suggestion = self.conn.experiments(e.id).suggestions().create()
+      self.conn.experiments(e.id).observations().create(
+        suggestion=suggestion.id,
+        values=[dict(name="y1", value=numpy.random.rand()), dict(name="y2", value=numpy.random.rand())],
+      )
 
-    best_assignments_logger = BestAssignmentsLogger(experiment)
-    best_observations_from_logger = best_assignments_logger.fetch(observations)
-    self.assert_observations_are_sorted(best_observations_from_logger, experiment)
+    best_assignments = self.conn.experiments(e.id).best_assignments().fetch()
+    best_assignments_list = list(best_assignments.iterate_pages())
+    self.assert_best_assignments_are_sorted(e, best_assignments_list)
+    assert len(best_assignments_list) > 1
 
-    for best_observation in best_observations_from_logger:
-      for i, metric in enumerate(best_observation["values"]):
+    for best_assignment in best_assignments_list:
+      for i, value in enumerate(best_assignment.values):
         if objectives[i] == "maximize":
-          assert metric["value"] >= thresholds[i]
+          assert value.value >= thresholds[i]
         if objectives[i] == "minimize":
-          assert metric["value"] <= thresholds[i]
```

Comment:
you can remove `thresholds` and `objectives` and just loop over the metrics here:
```
for best_assignment in best_assignments_list:
  for i, metric in enumerate(experiment_meta["metrics"]):
    assert metric["name"] == best_assignment.values[i].name
    if metric["objective"] ....
```
also it makes explicit that the order of metric values should be the same as the experiment.metrics